internal/cmd/controller/reconciler/cluster_controller.go

```diff
@@ -94,15 +94,20 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if err != nil {
 		return ctrl.Result{}, r.updateErrorStatus(ctx, req.NamespacedName, cluster.Status, err)
 	}
+
+	deleted := map[types.UID]bool{}
 	for _, bundle := range cleanup {
-		for _, bundleDeployment := range bundleDeployments.Items {
-			if bundleDeployment.Labels[fleet.BundleLabel] == bundle.Name &&
-				bundleDeployment.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace {
-				logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bundleDeployment)
-				err := r.Delete(ctx, &bundleDeployment) // nolint:gosec // does not store pointer
+		for _, bd := range bundleDeployments.Items {
+			if deleted[bd.GetUID()] {
+				continue
+			}
+			if bd.Labels[fleet.BundleLabel] == bundle.Name && bd.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace {
+				logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bd)
+				err := r.Delete(ctx, &bd) // nolint:gosec // does not store pointer
```

Comment:
Nit: I THINK we don't need this `// nolint` statement?. (I known it's not part of the changes of the PR, but just saw it and remembered that I've seen it in other files) 
I've tested running `golangci-lint` locally without the `// nolint` and it didn't complain.

Reply:
Ah, nice. Last time I checked I still got warnings. I'll add a commit, but leave the ones in gitrepo controller to you, so we don't conflict.